src/lambda/eval.rs

```diff
@@ -0,0 +1,28 @@
+use core::future::Future;
+use std::pin::Pin;
+
+use anyhow::Result;
+
+use super::{Concurrency, EvaluationContext, ResolverContextLike};
+
+pub trait Eval<Output = async_graphql::Value>
+where
+  Self: Send + Sync,
+{
+  fn eval<'a, Ctx: ResolverContextLike<'a> + Sync + Send>(
+    &'a self,
+    ctx: &'a EvaluationContext<'a, Ctx>,
+    conc: &'a Concurrency,
+  ) -> Pin<Box<dyn Future<Output = Result<Output>> + 'a + Send>>
+  where
+    Output: 'a,
+  {
+    Box::pin(self.async_eval(ctx, conc))
+  }
+
+  fn async_eval<'a, Ctx: ResolverContextLike<'a> + Sync + Send>(
+    &'a self,
+    ctx: &'a EvaluationContext<'a, Ctx>,
+    conc: &'a Concurrency,
+  ) -> impl Future<Output = Result<Output>> + Send;
```

Comment:
We should keep only one implementation or make the return type an associated type for Eval.